includes/class-wc-stripe-blocks-support.php

```diff
@@ -354,39 +354,38 @@ public function add_payment_request_order_meta( PaymentContext $context, Payment
 		}
 
 		$is_stripe_payment_method = $this->name === $context->payment_method;
+		$main_gateway             = WC_Stripe::get_instance()->get_main_stripe_gateway();
+		$is_upe                   = $main_gateway instanceof WC_Stripe_UPE_Payment_Gateway;
 
 		// Check if the payment method is a UPE payment method. UPE methods start with `stripe_`.
-		if ( ! $is_stripe_payment_method && 0 === strpos( $context->payment_method, "{$this->name}_" ) ) {
-			$main_gateway = WC_Stripe::get_instance()->get_main_stripe_gateway();
-
-			// If the main gateway is not a UPE gateway, then we don't need to do anything.
-			if ( $main_gateway instanceof WC_Stripe_UPE_Payment_Gateway ) {
-				// Strip "Stripe_" from the payment method name to get the payment method type.
-				$payment_method_type      = substr( $context->payment_method, strlen( $this->name ) + 1 );
-				$is_stripe_payment_method = isset( $main_gateway->payment_methods[ $payment_method_type ] );
-
-				/**
-				 * When using the block checkout and a saved token is being used, we need to set a flag
-				 * to indicate that deferred intent should be used.
-				 */
-				if ( $is_stripe_payment_method && isset( $data['issavedtoken'] ) && $data['issavedtoken'] ) {
-					$context->set_payment_data( array_merge( $data, [ 'wc-stripe-is-deferred-intent' => true ] ) );
-				}
-			}
+		if ( $is_upe && ! $is_stripe_payment_method && 0 === strpos( $context->payment_method, "{$this->name}_" ) ) {
+			// Strip "Stripe_" from the payment method name to get the payment method type.
+			$payment_method_type      = substr( $context->payment_method, strlen( $this->name ) + 1 );
+			$is_stripe_payment_method = isset( $main_gateway->payment_methods[ $payment_method_type ] );
+		}
+
+		if ( ! $is_stripe_payment_method ) {
+			return;
+		}
+
+		/**
+		 * When using UPE on the block checkout and a saved token is being used, we need to set a flag
+		 * to indicate that deferred intent should be used.
+		 */
+		if ( $is_upe && isset( $data['issavedtoken'] ) && $data['issavedtoken'] ) {
```

Comment:
I ended up refactoring this function a little bit so that we are able to add the `wc-stripe-is-deferred-intent` data when the payment method is just 'stripe' and the store has UPE enabled.

Not sure if these changes make the function harder to understand, but I'm open to any other suggestions to improve this :)